client/my-sites/checkout/checkout/emergent-paywall-box.jsx

```diff
@@ -242,5 +242,7 @@ export default connect(
 	state => ( {
 		userCountryCode: getCurrentUserCountryCode( state ),
 	} ),
-	null
+	{
+		recordTracksEvent,
+	}
```

Comment:
further we can encode a more semantic meaning than just "track event" and remove the side-effects from the React component

```js
dispatch => ( {
	onOrderCreated: ( { orderId, paymentMethod, siteSlug } ) => {
		dispatch( recordTracksEvent( 'calypso_checkout_form_submit', { payment_method: paymentMethod } ) );
		page.redirect( `/checkout/thank-you/${ siteSlug }/pending/{ orderId }` );
	}
} )
```

Reply:
I see what you did there... :) Have a 🌮

Reply:
ideally I would love to see that because in addition to separating out routing concerns from view concerns it would free us up more to change the routing system (a long-term low-priority task of ours)

that being said in most cases practicality wins over the long-term goal and people put it in. for analytics and routing I'd recommend this same advice: if it's simple to extract then do so; if it's awkward and cumbersome then try but be willing to leave it in the component.